modules/tutorial-advanced-minimalist-fulcro/pages/index.adoc

```diff
@@ -0,0 +1,338 @@
+    (ui-my-router router)))
+
+(defn init [app]
+  ;; Avoid startup async timing issues by pre-initializing things before mount
+  (app/set-root! app Root {:initialize-state? true})
+  (dr/initialize! app)                                         ; <4>
+  (run! #(merge/merge-component! app Person %
+           :append (conj (comp/get-ident AllPeople {}) :all-people))
+    [#:person{:id 100 :name "Kamča" :biography "..."}
+     #:person{:id 123 :name "Doubravka" :biography "..."}])
+  (dr/change-route! app ["person" "123"])                       ; <5>
+  ;; Note: 👆🏿 will warn https://book.fulcrologic.com/#warn-uism-sm-not-in-state; ignore! <7>
+  ;; or: (dr/change-route! app (dr/path-to Person "123"))           ; <6>
+  ;; or: (dr/change-route! app (dr/path-to Person {:person-id "123"})) ; <6>
+  (app/mount! app Root "app" {:initialize-state? false}))
+```
+====
+<1> You _must_ compose the router's query into its parent, under an arbitrary name (the `:ui/` ns is practical so that we do not unnecessarily include it in queries sent to the server)
+<2> You _must_ compose the router's initial state into its parent (notice we use the https://book.fulcrologic.com/#_template_mode[template mode of initial state] so the `{}` actually means "include the router's initial state")
+<3> We use `dr/change-route!` with the absolute route (concatenation of all route segments along the way from Root) to display the desired component
+<4> When we start the application, we need to initialize the dynamic routing subsystem with `dr/initialize!`
+<5> It is highly advisable to also make sure each router in the app is "routed" and thus in a well-defined state before it is ever displayed. Here we achieve that by routing to it in the init function. This can only be done after the application's state is initialized, i.e. either `app/set-root!` with `{:initialize-state? true}` or `app/mount!`
+    * link:{url-book}#_setting_the_route_early[The Fulcro book advises]: "`An even better approach is to use state to block rendering until such time as a route or load is ready just by looking at the current state of your top-most router’s state machine.`" - see the example code below the quote
+<6> Alternatively, instead of hard-coding the route as a vector of strings, we can use `(path-to & targets-and-params)` with component names to get the route - which is easier to navigate and to refactor. In the map form, there is just a single map with all the route's params at the very end.
+<7> Note: As of Fulcro 3.5.6, you are likely to get the https://book.fulcrologic.com/#warn-uism-sm-not-in-state[Attempt to get an ASM path...] error for the router because `dr/initialize!` is asynchronous and does not finish before the `dr/change-route!` call. You can safely ignore it.
+
+See the https://github.com/holyjak/minimalist-fulcro-template-backendless/blob/example/bare-essentials-routing/src/com/example/ui.cljs[full code here].
+
+=== Binding the route to the URL
```

Comment:
Hukka: After reading the part about binding to the URL, I'm wondering how I should model the typical structure where there is no specific "all" route, but an empty route means all. Perhaps there should be an explicit route name in the state, even if it's not in the URL?

JH: Correct. You would use something all `:route-segment ["all"]` and modify your URL<>route mapping code to replace "all" with "" when going to URL and vice versa.